chain/index/events_test.go

```diff
@@ -0,0 +1,412 @@
+package index
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	pseudo "math/rand"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+
+	"github.com/filecoin-project/lotus/chain/types"
+)
+
+func TestGetEventsForFilterNoEvents(t *testing.T) {
+	ctx := context.Background()
+	seed := time.Now().UnixNano()
+	t.Logf("seed: %d", seed)
+	rng := pseudo.New(pseudo.NewSource(seed))
+
+	headHeight := abi.ChainEpoch(60)
+	si, _, cs := setupWithHeadIndexed(t, headHeight, rng)
+	t.Cleanup(func() { _ = si.Close() })
+
+	// Create a fake tipset at height 1
+	fakeTipSet1 := fakeTipSet(t, rng, 1, nil)
+
+	// Set the dummy chainstore to return this tipset for height 1
+	cs.SetTipsetByHeightAndKey(1, fakeTipSet1.Key(), fakeTipSet1) // empty DB
+
+	// tipset is not indexed
+	f := &EventFilter{
+		MinHeight: 1,
+		MaxHeight: 1,
+	}
+	ces, err := si.GetEventsForFilter(ctx, f, false)
+	require.True(t, errors.Is(err, ErrNotFound))
+	require.Equal(t, 0, len(ces))
+
+	tsCid, err := fakeTipSet1.Key().Cid()
+	require.NoError(t, err)
+	f = &EventFilter{
+		TipsetCid: tsCid,
+	}
+
+	ces, err = si.GetEventsForFilter(ctx, f, false)
+	require.True(t, errors.Is(err, ErrNotFound))
+	require.Equal(t, 0, len(ces))
+
+	// tipset is indexed but has no events
+	err = withTx(ctx, si.db, func(tx *sql.Tx) error {
+		return si.indexTipset(ctx, tx, fakeTipSet1)
+	})
+	require.NoError(t, err)
+
+	ces, err = si.GetEventsForFilter(ctx, f, false)
+	require.NoError(t, err)
+	require.Equal(t, 0, len(ces))
+
+	f = &EventFilter{
+		TipsetCid: tsCid,
+	}
+	ces, err = si.GetEventsForFilter(ctx, f, false)
+	require.NoError(t, err)
+	require.Equal(t, 0, len(ces))
+
+	// search for a range that is absent
+	f = &EventFilter{
+		MinHeight: 100,
+		MaxHeight: 200,
+	}
+	ces, err = si.GetEventsForFilter(ctx, f, false)
+	require.NoError(t, err)
+	require.Equal(t, 0, len(ces))
+}
+
+func TestGetEventsForFilterWithEvents(t *testing.T) {
+	ctx := context.Background()
+	seed := time.Now().UnixNano()
+	t.Logf("seed: %d", seed)
+	rng := pseudo.New(pseudo.NewSource(seed))
+	headHeight := abi.ChainEpoch(60)
+	si, _, cs := setupWithHeadIndexed(t, headHeight, rng)
+	t.Cleanup(func() { _ = si.Close() })
+
+	ev1 := fakeEvent(
+		abi.ActorID(1),
+		[]kv{
+			{k: "type", v: []byte("approval")},
+			{k: "signer", v: []byte("addr1")},
+		},
+		[]kv{
+			{k: "amount", v: []byte("2988181")},
+		},
+	)
+
+	ev2 := fakeEvent(
+		abi.ActorID(2),
+		[]kv{
+			{k: "type", v: []byte("approval")},
+			{k: "signer", v: []byte("addr2")},
+		},
+		[]kv{
+			{k: "amount", v: []byte("2988181")},
+		},
+	)
+
+	events := []types.Event{*ev1, *ev2}
+
+	fm := fakeMessage(address.TestAddress, address.TestAddress)
+	em1 := executedMessage{
+		msg: fm,
+		evs: events,
+	}
+
+	si.SetActorToDelegatedAddresFunc(func(ctx context.Context, emitter abi.ActorID, ts *types.TipSet) (address.Address, bool) {
```

Comment:
this function isn't really doing anything, you may as well return `nil, false` I think, unless you want to test addresses

Reply:
At the very least, we need to return a valid ID address here for the event to get indexed. Let's keep this.